### PR TITLE
feat: add Vue.config configuration to nuxt.config.js

### LINKED ISSUE #2910: vue.config seems to be inaccessible from within nuxt
i couldn't find a way to configure Vue via vue.config before nuxt uses it.

https://vuejs.org/v2/api/

I need it for ignoring custom-elements. 
Is there any way to deal with the vue.config which i missed? 
I looked at the nuxt-source, but didn't find a place for that config.

<div align="right"><sub><em>This question is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt/nuxt.js/issues/c2527">#c2527</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by TheAlexLichter**
After #3564 and the request of #3427  (plus #3452) I decided to add a dedicated `vueConfig`object to the `nuxt.config.js` file.

The keys will be projected on Vue.config.

By default, `Vue.config.performance` is enabled in dev mode, `Vue.config.silent` is enabled in production mode.

Doc PR incoming.

Resolves: #2910, #3427

### CODE REVIEW

**Review by galvez — CHANGES_REQUESTED** [review]
👍 just found a hanging `console.log()`

**Review by clarkdo — COMMENTED** [review]
`Vue.config` can be easily set through `plugin` or `layout`, so is it necessary to embed in core ?
Maybe it's better to develop a module.

**Comment by manniL** [comment]
@clarkdo Well, it was [your suggestion](https://github.com/nuxt/nuxt.js/pull/3564#discussion_r204215280) :stuck_out_tongue: 

If the "*performance loss*" is negligible, why not? :)

File: lib/builder/builder.js [hunk 1]
```diff
@@ -182,6 +182,7 @@ export default class Builder {
       uniqBy: _.uniqBy,
       isDev: this.options.dev,
       debug: this.options.debug,
+      vue: this.options.vue,
```
Comment (clarkdo) [inline lib/builder/builder.js]: How about only pass `vue.config` into it ?

File: lib/app/client.js [hunk 2]
```diff
@@ -30,7 +30,10 @@ let router
 // Try to rehydrate SSR data from window
 const NUXT = window.__NUXT__ || {}
 
-Vue.config.silent = <%= !debug %>
+const vueConfigOptions = <%= serialize(vue.config) %>
+Object.keys(vueConfigOptions).forEach((option) => {
```
Comment (clarkdo) [inline lib/app/client.js]: How about using `Object.assgin` ?

```js
Object.assign(Vue,config, <%= serialize(vue.config) %>)
```

**Review by clarkdo — CHANGES_REQUESTED** [review]

File: lib/app/client.js (hunk 2 above)
Comment (manniL) [reply lib/app/client.js]: Yup, way better! :D

**Review by clarkdo — APPROVED** [review]

**Review by atinux — APPROVED** [review]

**Pull request merged**